scene/2d/animated_sprite_2d.cpp

```diff
@@ -172,17 +172,17 @@ void AnimatedSprite2D::_notification(int p_what) {
 				return;
 			}
 
-			double speed = frames->get_animation_speed(animation) * Math::abs(speed_scale);
-			if (speed == 0) {
-				return; // Do nothing.
-			}
-			int last_frame = frames->get_frame_count(animation) - 1;
-
 			double remaining = get_process_delta_time();
+			int i = 0;
 			while (remaining) {
-				if (timeout <= 0) {
-					timeout = _get_frame_duration();
+				if (speed_scale == 0) {
+					return; // Do nothing.
+				}
+
+				int fc = frames->get_frame_count(animation);
 
+				if (timeout <= 0) {
+					int last_frame = fc - 1;
```

Comment:
Again, changing the timeout check timing is fine, but no other changes are necessary in here.

Reply:
Moved getting the frame count into the loop because it can change after the `animation_finished` and `frame_changed` signal handlers. Alternatively, we can remove it here, but then I think we should add a warning in the signals documentation.

The same goes for `speed_scale` (plus one more reason below).

Reply:
It would be better to add a comment in the code why is it getting it in the while.
```cpp
// Frame count may be changed by animation_finished or frame_changed signals.
int fc = frames->get_frame_count(animation);
```

Reply:
Done.